express-zod-api/src/zts.ts

```diff
@@ -70,11 +70,16 @@ const onLiteral: Producer = ({ _zod: { def } }: $ZodLiteral) => {
   return values.length === 1 ? values[0] : f.createUnionTypeNode(values);
 };
 
-const onInterface: Producer = (int: $ZodInterface, { next, makeAlias }) =>
-  makeAlias(int, () => {
-    const members = Object.entries(int._zod.def.shape).map<ts.TypeElement>(
+const onObject: Producer = (
+  obj: $ZodObject,
+  { isResponse, next, makeAlias },
+) => {
+  const fn = () => {
+    const members = Object.entries(obj._zod.def.shape).map<ts.TypeElement>(
       ([key, value]) => {
-        const isOptional = int._zod.def.optional.includes(key);
+        const isOptional = isResponse
+          ? isSchema<$ZodOptional>(value, "optional")
+          : doesAccept(value, undefined);
```

Comment:
there is an util for that now
https://github.com/colinhacks/zod/pull/4074/commits/054b1be36809301105ba89c99ee2648d8f500620#diff-bf366bef5d66802671c79bde5d11d97168d1f8c5a6888ca30792cd521c4812b8R547

Reply:
made todo to fix later